src/EditorFeatures/Test2/Rename/InlineRenameTests.vb

```diff
@@ -1965,5 +1965,46 @@ End Class
                 Assert.Equal(InlineRenameFileRenameInfo.NotAllowed, session.FileRenameInfo)
             End Using
         End Sub
+
+        <WpfFact>
+        <WorkItem(28474, "https://github.com/dotnet/roslyn/issues/28474")>
+        <Trait(Traits.Feature, Traits.Features.Rename)>
+        Public Async Function HandleProjectsWithoutCompilations() As Task
+            Using workspace = CreateWorkspaceWithWaiter(
+                    <Workspace>
+                        <Project Language="C#" AssemblyName="CSharpProject" CommonReferences="true">
+                            <Document>
+                                public interface IGoo
+                                {
+                                    void [|$$Goo|]();
+                                }
+                                public class C : IGoo
+                                {
+                                    public void [|Goo|]() {}
+                                }
+                            </Document>
+                        </Project>
+                        <Project Language="NoCompilation" CommonReferences="false">
+                            <ProjectReference>CSharpProject</ProjectReference>
+                            <Document>
+                                // a no-compilation document
+                            </Document>
+                        </Project>
+                    </Workspace>)
+
+                Dim session = StartSession(workspace)
+
+                ' Type a bit in the file
```

Comment:
Why does this need a fully blown test with the editors? If you look in https://github.com/dotnet/roslyn/blob/master/src/EditorFeatures/Test2/Rename/RenameEngineTests.vb there's a bunch of tests that are just "rename Goo to Bar" tests are the usual case. This file is normally intended for stuff specific to the inline rename experience.

As a comparison, think of this as being like the completion command handler tests: sometimes you need to test that, but if you're just trying to assert that an item is shown you test the completion provider directly.

Reply:
don't know, don't care :)  i just wanted a test that validates the case :)   i didn't have a lot of time to do this PR, and i was sad there were customers complaining about hte scenario.

Reply:
Pushed the conversion over to your branch. Will merge once CI runs.

Reply:
> Pushed the conversion over to your branch. Will merge once CI runs.

Thanks!  IMO, we should keep both tests.  There's enough interesting logic in both places that i think it's worth covering our bases.  Up to you!